lib/go/thrift/binary_protocol.go

```diff
@@ -432,6 +432,15 @@ func (p *TBinaryProtocol) ReadString(ctx context.Context) (value string, err err
 		err = invalidDataLength
 		return
 	}
+	if size == 0 {
+		return "", nil
+	}
+	if size < int32(len(p.buffer)) {
+		// Avoid allocation on small reads
+		buf := p.buffer[:size]
+		read, e := io.ReadFull(p.trans, buf)
+		return string(buf[:read]), NewTProtocolException(e)
+	}
```

Comment:
why not move this logic inside `readStringBody` and use that from both protocols?

Reply:
This avoids the extra allocation (we reused `p.buffer`)

`readStringBody` is a function attached to type `TBinaryProtocol`, so it cannot be used by `TCompactProtocol`. we cannot move it to top level function either as it's also used by other parts of `TBinaryProtocol`. I mean we can do that refactor, but consider it's just 4 lines it's not really worth it.